lib/rubygems/config_file.rb

```diff
@@ -210,22 +210,32 @@ def initialize(args)
       @hash = @hash.merge environment_config
     end
 
+    @hash.transform_keys! do |k|
+      if %w[backtrace bulk_threshold verbose update_sources cert_expiration_length_days
+            install_extension_in_lib ipv4_fallback_enabled].include?(k)
```

Comment:
How did you choose the list of configuration keys? I see there are a few more than this.

Reply:
Make sensed. It's not strong reason.

I update that to deny list.